apps/prairielearn/src/middlewares/authzHasCourseInstanceAccess.tsx

```diff
@@ -0,0 +1,49 @@
+import { type NextFunction, type Request, type Response } from 'express';
+
+import { HttpStatusError } from '@prairielearn/error';
+
+import { PageLayout } from '../components/PageLayout.js';
+import { getPageContext } from '../lib/client/page-context.js';
+import { Hydrate } from '../lib/preact.js';
+
+import { AuthzAccessMismatch } from './AuthzAccessMismatch.js';
+
+export default function (req: Request, res: Response, next: NextFunction) {
+  const effectiveAccess =
```

Comment:
Even though there are multiple ways to access this info (`res.locals.authn_...`, `res.locals. ...`, etc.) I chose this way as it was already there and felt the most self contained.